src/In1d.chpl

```diff
@@ -87,4 +89,25 @@ module In1d
         }
         return truth;
     }
+
+    // For each value in the first array, find the indices of their appearances
+    // in the second. Results are ordered by the order of the keys. 
+    proc indexof1d(keys: ?t, arr: t) throws {
+        var l : list(int, false);
+        var indexLists: [0..<keys.size] list(int, false);
+        
+        forall keyIdx in 0..<keys.size {
+            var keyVal = keys[keyIdx];
+            var indexList: list(int, false);
+            for i in 0..<arr.size do
+                if keyVal == arr[i] then indexList.pushBack(i);
+            indexLists[keyIdx] = indexList;
+        }
+
+        var finalList : list(int);
+        for l in indexLists do
+            finalList.pushBack(l);
+        
+        return finalList.toArray();
+    }
```

Comment:
Noting that we talked about the performance here; we don't expect it to perform well, but are focusing on correctness this first pass.